### ISSUE TITLE
Add support for Management Groups

### ISSUE THREAD

**Issue by markjgardner** [terraform-providers/terraform-provider-azurerm#1401]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support for [Management Groups](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management-groups-create) would greatly simply my current pattern for managing role and policy definitions that span multiple subscriptions. As opposed to having to create and manage an instance of each per subscription, I could create the definition once and apply it to the entire management group.

### New or Affected Resource(s)

New Resources: 

 * azurerm_management_group

Affected Resources: 

 * azurerm_policy_definition
 * azurerm_policy_assignment
 * azurerm_role_definition
 * azurerm_role_assignment

### Potential Terraform Configuration

```hcl

data "azurerm_client_config" "config" {}

data "azurerm_subscrition" "sub" {}

resource "azurerm_management_group" "group1" {
  name             = "MyManagementGroup"
  subscription_ids = [
    "00000000-0000-0000-0000-000000000000",
    "11111111-1111-1111-1111-111111111111",
    "22222222-2222-2222-2222-222222222222",
  ]
}

resource "azurerm_role_definition" "roledefinition1" {
  role_definition_id = "00000000-0000-0000-0000-000000000000"
  name               = "Role Definition"
  scope              = "${azurerm_management_group.group1.id}"
  description        = "Custom role"

  permissions {
    actions     = ["*"]
    not_actions = []
  }

  assignable_scopes = [
    "${data.azurerm_subscription.sub.id}", 
  ]
}

resource "azurerm_role_assignment" "roleassignment1" {
  scope                = "${azurerm_management_group.group1.id}"
  role_definition_name = "${azurerm_role_definition.roledefinition1.name}"
  principal_id         = "${data.azurerm_client_config.config.service_principal_object_id}"
}

resource "azurerm_policy_definition" "policy" {
  name         = "TestPolicy"
  policy_type  = "Custom"
  display_name = "Test policy definition"
  scope        = "${azurerm_management_group.group1.id}"
  policy_rule  = <<POLICY_RULE
    {
    "if": {
      "true"
    },
    "then": {
      "effect": "audit"
    }
  }
POLICY_RULE
}

resource "azurerm_policy_assignment" "test" {
  name                 = "example-policy-assignment"
  scope                = "${azurerm_management_group.group1.id}"
  policy_definition_id = "${azurerm_policy_definition.policy.id}"
  description          = "Policy Assignment"
  display_name         = "Test Policy Assignment"
}
```

### References

* #0000

**Comment by markjgardner** [comment]
I'd like to give this feature request a bump as Management Groups are now generally available: https://azure.microsoft.com/en-us/roadmap/azure-management-groups-now-available/

**Comment by justinbarias** [comment]
I've gone and started work on this myself.. Just writing the tests now.. but so far looks good.. Don't support nested management groups yet.

**Comment by justinbarias** [comment]
PR submitted #1788

**Comment by tombuildsstuff** [comment]
Fixed via #1788 - thanks @justinbarias :)

**Comment by tombuildsstuff** [comment]
hey @markjgardner @justinbarias 

Just to let you know that this has been released in v1.14.0 of the AzureRM Provider which is now available: https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.14.0/CHANGELOG.md

Thanks!
